resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala

```diff
@@ -241,14 +241,14 @@ class KubernetesConfSuite extends SparkFunSuite {
   test("SPARK-36566: get app name label") {
     assert(KubernetesConf.getAppNameLabel(" Job+Spark-Pi 2021") === "job-spark-pi-2021")
     assert(KubernetesConf.getAppNameLabel("a" * 63) === "a" * 63)
-    assert(KubernetesConf.getAppNameLabel("a" * 64) === "a" * 63)
-    assert(KubernetesConf.getAppNameLabel("a" * 253) === "a" * 63)
+    assert(KubernetesConf.getAppNameLabel("a" * 64) === "a" * 62)
```

Comment:
Is this correct?

Reply:
When `KUBERNETES_DNS_LABEL_NAME_MAX_LENGTH` is 63, why do we need to get `62`? The previous result looks more correct to me. Could you explain your rational?

Reply:
The result is correct, because `KubernetesConf#getAppNameLabel` calls `.stripSuffix("-")` after `abbreviate`

```
scala> StringUtils.abbreviate("aaaa", "-", 3)
res4: String = aa-

scala> StringUtils.abbreviate("aaaa", "", 3).stripPrefix("-").stripSuffix("-")
res5: String = aaa
```

Reply:
- `64 * a` --(abbreviate)--> `62 * a + '-'` --(strip)--> `62 * a`
- `63 * a` --(abbreviate)--> `63 * a` --(strip)--> `63 * a`

If we want to apply this fix, we'd bettter to expalin the behavior, we might need to doc some on `getAppNameLabel` (or test at least), the current note will make people think that it is simply taking the first 63 characters. but current behavior is：

- If len <= 63, return dircetly
- If len > 63, return first 62 characters.

Reply:
Agreed with @Yikun , and comments should be added in the code to clarify the change intent, so as to avoid being wrongly changed by other developers in the future ...

Reply:
Emm, how about using "-abbr" instead of "-"?

Reply:
To avoid confusing, I changed the `abbrevMarker` to "-abbr"